src/osp/Active/SysAreaAssociate.cpp

```diff
@@ -26,157 +26,162 @@
-//        {
-//            //std::cout << "already loaded\n";
-//            continue;
-//        }
-
-
-        // ignore self (unreachable as active areas are unloadable)
-        //if (&sat == m_sat)
-        //{
-        //    continue;
-        //}
-
-        // Calculate distance to satellite
-        // For these long vectors, 1024 units = 1 meter
-        //Vector3s relativePos = m_universe.sat_calc_position(m_areaSat, sat);
-        //Vector3 relativePosMeters = Vector3(relativePos) / 1024.0f;
-
-        //using Magnum::Math::pow;
-
-        // ignore if too far
-        //if (relativePos.dot() > pow(sat.get_load_radius()
-        //    + m_sat->get_load_radius(), 2.0f))
-        //{
-        //    continue;
-        //}
-
-        //std::cout << "is near! dist: " << relativePos.length() << "/"
-        //<< (sat.get_load_radius() + m_sat->get_load_radius()) << "\n";
-
-        // Satellite is near! Attempt to load it
-
-        sat_activate(sat);
+        tracker.m_enter.clear();
+        tracker.m_leave.clear();
     }
 
-}
+    // Deal with activating satellites that have a UCompActivationRadius
+    // Satellites that have an Activation Radius have a sphere around them. If
+    // this sphere overlaps the ActiveArea's sphere 'm_areaRadius', then it
+    // will be activated.
 
-void SysAreaAssociate::connect(universe::Satellite sat)
-{
-    // do more stuff here eventually
-    m_areaSat = sat;
-}
+    auto viewActRadius = rUni.get_reg().view<
+            UCompActivationRadius, UCompActivatable, UCompType>();
 
-void SysAreaAssociate::disconnect()
-{
-    if (!m_activatedSats.empty())
+    for (Satellite sat : viewActRadius)
     {
-        // De-activate all satellites
-
-        auto viewAct = m_scene.get_registry().view<ACompActivatedSat>();
+        auto satRadius = viewActRadius.get<UCompActivationRadius>(sat);
+        auto satType = viewActRadius.get<UCompType>(sat);
 
-        for (ActiveEnt ent : viewAct)
+        if (satType.m_type == TypeSatIndex::Invalid)
         {
-            sat_deactivate(ent, viewAct.get(ent));
+            continue; // Skip satellites that don't have a type (somehow)
         }
 
+        ActivationTracker &activations = pArea->m_activated[
+                                                    size_t(satType.m_type)];
+
+        // Check if already activated
+        auto found = activations.m_inside.find(sat);
+        bool alreadyInside = (found != activations.m_inside.end());
+
+        // Simple sphere-sphere-intersection check
+        float distanceSquared = rUni.sat_calc_pos_meters(areaSat, sat).dot();
+        float radius = areaUComp.m_areaRadius + satRadius.m_radius;
 
+        if ((radius * radius) > distanceSquared)
+        {
+            if (alreadyInside)
+            {
+                continue; // Satellite already activated
+            }
+            // Satellite is nearby, activate it!
+            auto entered = activations.m_inside.emplace(sat, entt::null).first;
```

Comment:
might be better to use C++17 structured bindings here.

Is m_inside a map? i assume it is

auto const& [iter, bool] = activations.m_inside.emplace(sat, entt::null);
activations.m_enter.emplace_back(iter);